src/extraction/manifest.py

```diff
@@ -115,6 +120,58 @@ def add_datafiles(  # pylint: disable=missing-function-docstring
     ) -> None:
         self._datafiles.extend(datafiles)
 
+    def serialize(self, root_dir: Path) -> None:
+        root_dir.mkdir(parents=True, exist_ok=True)
+
+        def serialize_objects(objects: Sequence[BaseModel], dir_name: str) -> None:
+            objects_dir = root_dir / dir_name
+            objects_dir.mkdir()
+
+            for i, obj in enumerate(objects):
+                file_path = (objects_dir / str(i)).with_suffix(".json")
+                with file_path.open("w", encoding="utf-8") as f:
+                    f.write(obj.model_dump_json(by_alias=True))
+
+        serialize_objects(self.get_projects(), "projects")
+        serialize_objects(self.get_experiments(), "experiments")
+        serialize_objects(self.get_datasets(), "datasets")
+        serialize_objects(self.get_datafiles(), "datafiles")
+
+    @staticmethod
+    def deserialize(root_dir: Path) -> IngestionManifest:
+        try:
+            directory = DirectoryNode(root_dir)
+        except NotADirectoryError as e:
+            e.strerror = f"Failed to deserialize ingestion manifest: {e.strerror}"
+            raise e
+
+        def deserialize_objects(
+            obj_dir: DirectoryNode, object_type: Type[ModelT]
+        ) -> list[ModelT]:
+            objects = []
+
+            json_files = obj_dir.find_files(lambda p: p.extension() == ".json")
+
+            for file in json_files:
+                with file.path().open("r", encoding="utf-8") as f:
+                    json_data = f.read()
+                    obj = object_type.model_validate_json(json_data)
+                    objects.append(obj)
+
+            return objects
+
+        projects = deserialize_objects(directory.dir("projects"), RawProject)
```

Comment:
Stylistic question rather than a comment that needs addressing, why serialize to a set of sub-directories rather than putting everything into one large JSON?

Reply:
Good question, I did consider that, but I thought we might end up with some very large JSON files in that case. For example, in some cases with the ABI data we will have thousands of datafiles, so one big JSON file would get into the tens of thousands of lines

Reply:
Makes sense you don't want the thing that tells you if the ingest has failed half way through to fail half way through because it ran out of memory trying to write a gigantic file or something.